### PR TITLE
Student facting i18n

### PR DESCRIPTION
**Pull request by Hamms**

### CODE REVIEW

File: bin/i18n/sync-codeorg-in [hunk 1]
```diff
@@ -7,57 +7,106 @@
...
+    end
+    temp_file.close
+    FileUtils.mv(temp_file.path, "dashboard/config/locales/#{filename}.en.yml")
   end
-  temp_file.close
-  FileUtils.mv(temp_file.path, "dashboard/config/locales/instructions.en.yml")
 end
 
 # Pull in instructions for levelbuilder levels from .level files
 # and save them to instructions.en.yml to be translated
```
Comment (tanyaparker) [inline bin/i18n/sync-codeorg-in]: Want to update this comment and maybe method name from `localize_instructions` to `localize_level_content`?

File: bin/i18n/sync-codeorg-in [hunk 2]
```diff
@@ -7,57 +7,106 @@
...
+    f.write(({ "en" => { "data" => { label => data }}}).to_yaml)
     File.write(f, File.read(f))
   end
 end
 
 def reformat_quotes
-  temp_file = Tempfile.new("tempinstructions.yml")
-  File.open("dashboard/config/locales/instructions.en.yml", "r") do |f|
-    f.each_line { |line| temp_file.puts line.gsub("'\"", '"').gsub("\"'", '"').gsub("''", "'")}
+  filenames = ["instructions", "markdown_instructions"]
```
Comment (tanyaparker) [inline bin/i18n/sync-codeorg-in:29]: I didn't think about this before, but if the markdown strings are in a .yml file Crowdin might not expect it to be markdown and may not process it the same. For example, they usually highlight code as green as an indicator to translators that it shouldn't be translated. I'm not sure if it'll be able to recognize the same here, but we'll test it out!

File: bin/i18n/utils.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'open3'
+
+def run_bash_script(location)
```
Comment (tanyaparker) [inline bin/i18n/utils.rb:3]: Nice work around 👍

File: bin/i18n/utils.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  Open3.popen3("sh #{location}") do |_stdin, stdout, _stderr, _wait_thread|
```
Comment (tanyaparker) [inline bin/i18n/utils.rb:4]: What does it mean if it starts with a `_` or not?

File: bin/i18n/utils.rb (hunk 3 above)
Comment (Hamms) [reply bin/i18n/utils.rb:3]: plus as a bonus it'll now pipe the output in realtime; no more waiting around without knowing how far along it is.

File: bin/i18n/utils.rb (hunk 4 above)
Comment (Hamms) [reply bin/i18n/utils.rb:4]: just ruby convention for denoting unused variables.

File: bin/i18n/sync-codeorg-in (hunk 2 above)
Comment (Hamms) [reply bin/i18n/sync-codeorg-in:29]: Interesting. That could be good or bad, as there are still a few markdown instructions that have raw HTML in them and which we would probably not want to be translated. I figure that once we get it all in crowdin and see what it looks like, we can have the sanitization conversation.

**Pull request merged**